x-pack/plugins/licensing/server/plugin.ts

```diff
@@ -4,145 +4,185 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import { Observable } from 'rxjs';
-import { first, map } from 'rxjs/operators';
+import { Observable, Subject, Subscription, merge, timer } from 'rxjs';
+import { take, takeUntil } from 'rxjs/operators';
 import moment from 'moment';
+import { createHash } from 'crypto';
+
 import {
   CoreSetup,
   CoreStart,
   Logger,
-  Plugin as CorePlugin,
+  Plugin,
   PluginInitializerContext,
+  IClusterClient,
 } from 'src/core/server';
-import { Poller } from '../../../../src/core/utils/poller';
-import { LicensingConfigType, LicensingPluginSetup, ILicense } from './types';
-import { LicensingConfig } from './licensing_config';
-import { License } from './license';
+
+import { ILicense, LicensingPluginSetup, PublicLicense, PublicFeatures } from '../common/types';
+import { License } from '../common/license';
+import { createLicenseUpdate } from '../common/license_update';
+
+import { ElasticsearchError, RawLicense, RawFeatures } from './types';
+import { LicenseConfigType } from './licensing_config';
 import { createRouteHandlerContext } from './licensing_route_handler_context';
 
-declare module 'src/core/server' {
-  interface RequestHandlerContext {
-    licensing: {
-      license: ILicense;
+function normalizeServerLicense(license: RawLicense): PublicLicense {
+  return {
+    uid: license.uid,
+    type: license.type,
+    expiryDateInMillis: license.expiry_date_in_millis,
+    status: license.status,
+  };
+}
+
+function normalizeFeatures(rawFeatures: RawFeatures) {
+  const features: PublicFeatures = {};
+  for (const [name, feature] of Object.entries(rawFeatures)) {
+    features[name] = {
+      isAvailable: feature.available,
+      isEnabled: feature.enabled,
     };
   }
+  return features;
 }
 
-export class Plugin implements CorePlugin<LicensingPluginSetup> {
+function sign({
+  license,
+  features,
+  error,
+}: {
+  license?: PublicLicense;
+  features?: PublicFeatures;
+  error?: string;
+}) {
+  return createHash('md5')
```

Comment:
I know it's somewhat unrelated to the migration effort, but since we're here, what do you think about changing the algorithm to `SHA-256`? Some of our ongoing security initiatives may prevent us from being able to use `md5`.

Reply:
np, we discussed it with @azasypkin that `md5` can de deprecated. It there a list of approved encryption algorithms?

Reply:
Thanks!

> It there a list of approved encryption algorithms?

"It depends" :). There isn't any one list, but if we wanted to run Kibana in say, a FIPS environment, then we'd need to abide by [this list](https://csrc.nist.gov/csrc/media/publications/fips/140/2/final/documents/fips1402annexa.pdf).